src/ucp/core/ucp_worker.c

```diff
@@ -1918,17 +1923,34 @@ ucp_worker_add_rkey_config(ucp_worker_h worker, const ucp_rkey_config_key_t *key
         goto err;
     }
 
-    /* initialize rkey configuration */
+    ucs_assert((key->sys_dev == UCS_SYS_DEVICE_ID_UNKNOWN) ||
+               (lanes_distance != NULL));
+
+    /* Initialize rkey configuration */
     rkey_cfg_index   = worker->rkey_config_count;
     rkey_config      = &worker->rkey_config[rkey_cfg_index];
     rkey_config->key = *key;
-    status           = ucp_proto_select_init(&rkey_config->proto_select);
+
+    /* Copy lanes distance to remote memory */
```

Comment:
why is it remote memory? did you mean remote key?

Reply:
reworded